File: console/cpp/binary-ante/src/Deck.cpp
#include "Deck.h"

using namespace std;


Deck::Deck(BinaryType type){
    deck = binary(type);
    top = binary(type-1);
}
Deck::~Deck(){
}


void Deck::print(){
    cout << "Deck: (10)" << deck.toDec() << " = (2)" << deck.toString() << "\tTop: (10)" << top.toDec() << " = (2)" << top.toString() << endl;
    cout << endl;
}

File: console/CSharp/binary-ante/Deck.cs
using System;
using VT.Types.Binary;

namespace VT.BinaryAnte
{
	public class Deck
	{
		private Binary deck, top;


		public Deck (BinaryType type){
			deck = new Binary(type);
			top = new Binary((BinaryType)((int)(type)-1));
		}
	}
}

